packages/plugin-ext/src/plugin/env.ts

```diff
@@ -29,11 +30,20 @@ export abstract class EnvExtImpl {
     private ui: theia.UIKind;
     private envMachineId: string;
     private envSessionId: string;
+    private host: string;
+    private _isTelemetryEnabled: boolean;
+    private _remoteName: string | undefined;
+    private onDidChangeTelemetryEnabledEmitter: Emitter<boolean>;
 
     constructor(rpc: RPCProtocol) {
         this.proxy = rpc.getProxy(PLUGIN_RPC_CONTEXT.ENV_MAIN);
         this.envSessionId = v4();
         this.envMachineId = v4();
+        this.host = 'desktop';
```

Comment:
Suggestion: It looks to me like [vscode](https://github.com/microsoft/vscode/blob/edc585edf6ec34d121196a3eda229c1c0fed676e/src/vs/workbench/services/extensions/browser/webWorkerExtensionHost.ts#L280) uses `web` here when the `uiKind` is also `web`. Since we set `uiKind` to `web` when running in the browser mode, we should probably also set `host` to `web` in that case.

Reply:
I understood `web` as in the extension host would be running in a browser? And `desktop` would be more similar to how VS Code desktop runs: the plugin host is a regular node process. Not sure what's the actual semantic for this value.

Reply:
I see, makes sense.